src/local-history/local-history-manager.ts

```diff
@@ -530,4 +535,17 @@ export class LocalHistoryManager {
         await vscode.env.clipboard.writeText(revisionFolderPath);
     }
 
+    /**
+     * Modifies the permission of a file to 'WRITE'.
+     * @param uri the path of source file.
+     */
+    private modifyFilePermissions(uri: string | vscode.Uri) {
+        //Change the permission of the file to 'WRITE'.
+        fs.chmod(typeof uri === 'string' ? uri : uri.fsPath, 0o200, (err) => {
```

Comment:
Do you mind updating the `write` file extracting the part where we change the permission as well?